### PR TITLE
[GEOS-10027] CoverageReaderFileConverter fails to convert valid input file path or url

### PR DESCRIPTION
**Pull request by taba90**
[GEOS-10027](https://osgeo-org.atlassian.net/browse/GEOS-10027) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geoserver&utm_campaign=badge_info)

Jira ticket https://osgeo-org.atlassian.net/browse/GEOS-10027

fix a regression caused  by the CoverageReaderFileConverter preventing a coverage file name to be retrieved when saving the store and automatically set as the layer name.

Before the introduction of the CoverageReaderFileConverter [the check on the input string](https://github.com/geoserver/geoserver/blob/2.17.x/src/main/src/main/java/org/geoserver/catalog/ResourcePool.java#L1645), 
 was returning false only if the URI was not a file uri, but was true even if the URI constructor failed.
This pr switch to the URL constructor that seems more tolerant  of the URI on the string passed, and performs few more checks.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.**

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the `main` branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

**Comment by taba90** [comment]
Docs builds looks unrelated since this pr doesn't provide any documentation change.

File: src/main/src/test/java/org/geoserver/catalog/ResourcePoolTest.java [hunk 1]
```diff
@@ -890,4 +892,53 @@ public void testCoverageReaderInputConverter() throws IOException {
...
+     * Tests that even if the input string cannot be parsed as a valid URI reference, but the it is
+     * nontheless a valid file URL, the CoverageReaderFileConverter is able to do the conversion.
+     *
+     * @throws IOException
+     */
+    @Test
+    public void testCoverageReaderInputConverterInvalidURI() throws IOException {
+        Assume.assumeTrue(SystemUtils.IS_OS_WINDOWS);
+        String fileURL =
+                MockData.class.getResource("tazdem.tiff").getFile().replaceAll("/", "\\\\");
```
Comment (aaime) [inline src/main/src/test/java/org/geoserver/catalog/ResourcePoolTest.java:906]: Could be a little simpler: 

```suggestion
                MockData.class.getResource("tazdem.tiff").getFile().replace('/', '\\');
```

File: src/main/src/main/java/org/geoserver/catalog/CoverageReaderFileConverter.java [hunk 2]
```diff
@@ -74,14 +76,21 @@ public CoverageReaderFileConverter(Catalog catalog) {
...
+        // Check to see if our "url" points to a file or not.
         try {
-            URI uri = new URI(input);
-            return uri.getScheme() == null || "file".equalsIgnoreCase(uri.getScheme());
-        } catch (URISyntaxException e) {
-            return false;
+            URI uri = new URL(input).toURI();
+            canConvert = uri.getScheme() == null || "file".equalsIgnoreCase(uri.getScheme());
+        } catch (MalformedURLException | URISyntaxException | IllegalArgumentException e) {
+            if (input.startsWith("file")) {
```
[the remark this answers is not in the document]
Comment (taba90) [reply src/main/src/main/java/org/geoserver/catalog/CoverageReaderFileConverter.java]: fixed

File: src/main/src/test/java/org/geoserver/catalog/ResourcePoolTest.java (hunk 1 above)
Comment (taba90) [reply src/main/src/test/java/org/geoserver/catalog/ResourcePoolTest.java]: fixed

**Comment by taba90** [comment]
thanks for your review @aaime feedbacks have been applied

**Comment by aaime** [comment]
Merged, thanks!

**Pull request merged**